--- a/examples/fpcalc.c
+++ b/examples/fpcalc.c
@@ -17,8 +17,10 @@
 #define MAX(a, b) ((a) > (b) ? (a) : (b))
 #define MIN(a, b) ((a) < (b) ? (a) : (b))
 
-#if LIBAVCODEC_VERSION_INT < AV_VERSION_INT(54, 28, 0)
-#define avcodec_free_frame av_freep
+#if LIBAVCODEC_VERSION_INT > AV_VERSION_INT(56, 41, 100)
+#define avcodec_free_frame av_frame_free
+#define avcodec_alloc_frame av_frame_alloc
+#define avcodec_get_frame_defaults av_frame_unref
 #endif
 
 int decode_audio_file(ChromaprintContext *chromaprint_ctx, const char *file_name, int max_length, int *duration)
mock -r fedora-23-i386-rpmfusion_free --no-clean rebuild
